src/VisualStudio/Core/Def/ProjectSystem/VisualStudioProjectFactory.cs

```diff
@@ -48,14 +45,14 @@ internal sealed class VisualStudioProjectFactory : IVsTypeScriptVisualStudioProj
             [ImportMany] IEnumerable<Lazy<IDynamicFileInfoProvider, FileExtensionsMetadata>> fileInfoProviders,
             HostDiagnosticUpdateSource hostDiagnosticUpdateSource,
             IVisualStudioDiagnosticAnalyzerProviderFactory vsixAnalyzerProviderFactory,
-            SVsServiceProvider serviceProvider)
+            IVsService<SVsSolution, IVsSolution2> solution2)
```

Comment:
Some tests are now failing with:

> Microsoft.VisualStudio.Composition.CompositionFailedException : Expected 1 export(s) with contract name "Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.VisualStudioProjectFactory" but found 0 after applying applicable constraints.

Is there some other asset or mock that I need to change to match the new signature of the importing constructor?

Reply:
We generally have a service provider mock here: https://github.com/dotnet/roslyn/blob/main/src/VisualStudio/TestUtilities2/MockServiceProvider.vb. I don't think we've ever actually used the `IVsService<>` interface before (this is news to me!) so this might require updating our mocks to support it.

Reply:
I didn't see anything around `MockServiceProvider` or along the callstack leading up to its creation that did anything other than reference the type for `VisualStudioProjectFactory`.  Nothing that looked like it was specifically referencing or mocking the importing constructor.

Reply:
Our tests actually create a MEF container and then pull things out of it: https://github.com/dotnet/roslyn/blob/b76cd4524d29ef01b28a68e63eb3a39a9215e363/src/VisualStudio/TestUtilities2/ProjectSystemShim/Framework/TestEnvironment.vb#L92

Reply:
You'll want to create an implementation of `IVsService<SVsSolution, IVsSolution2>` similar to this:
https://github.com/dotnet/roslyn/blob/421d20f47a82f6feb6c0a628387d2c14dde57850/src/EditorFeatures/TestUtilities/StubVsEditorAdaptersFactoryService.cs#L21

Then add it to the Visual Studio test composition here:
https://github.com/dotnet/roslyn/blob/421d20f47a82f6feb6c0a628387d2c14dde57850/src/VisualStudio/TestUtilities2/VisualStudioTestCompositions.vb#L24

Reply:
➡️ Now resolved